packages/dds/task-manager/src/test/taskManager.spec.ts

```diff
@@ -58,163 +58,207 @@ describe("TaskManager", () => {
+            assert.ok(taskManager1.assignedTask(taskId), "Task manager 1 is not assigned task");
             assert.ok(taskManager2.queued(taskId), "Task manager 2 should be queued");
-            assert.ok(!taskManager2.haveTaskLock(taskId), "Task manager 2 should not have lock");
+            assert.ok(!taskManager2.assignedTask(taskId), "Task manager 2 should not be assigned");
 
             taskManager1.abandon(taskId);
             containerRuntimeFactory.processAllMessages();
-            await lockTaskP2;
+            const isAssigned2 = await volunteerTaskP2;
+            assert.ok(isAssigned2, "Should resolve true");
 
             assert.ok(!taskManager1.queued(taskId), "Task manager 1 should be queued");
-            assert.ok(!taskManager1.haveTaskLock(taskId), "Task manager 1 should not have lock");
+            assert.ok(!taskManager1.assignedTask(taskId), "Task manager 1 should not be assigned");
             assert.ok(taskManager2.queued(taskId), "Task manager 2 should be queued");
-            assert.ok(taskManager2.haveTaskLock(taskId), "Task manager 2 should not have lock");
+            assert.ok(taskManager2.assignedTask(taskId), "Task manager 2 should not be assigned");
+        });
+
+        it("Can subscribe to a task", async () => {
+            const taskId = "taskId";
+            taskManager1.subscribeToTask(taskId);
+
+            assert.ok(taskManager1.queued(taskId), "Task manager 1 should be queued");
+            assert.ok(!taskManager1.assignedTask(taskId), "Task manager 1 should not be assigned");
+
+            containerRuntimeFactory.processAllMessages();
+
+            assert.ok(taskManager1.queued(taskId), "Task manager 1 should be queued");
+            assert.ok(taskManager1.assignedTask(taskId), "Task manager 1 is not assigned task");
+        });
+
+        it("Can subscribe and wait for a task", async () => {
+            const taskId = "taskId";
+            taskManager1.subscribeToTask(taskId);
+            taskManager2.subscribeToTask(taskId);
+
+            assert.ok(taskManager1.queued(taskId), "Task manager 1 should be queued");
+            assert.ok(!taskManager1.assignedTask(taskId), "Task manager 1 should not be assigned");
+            assert.ok(taskManager2.queued(taskId), "Task manager 2 should be queued");
+            assert.ok(!taskManager2.assignedTask(taskId), "Task manager 2 should not be assigned");
+
+            containerRuntimeFactory.processAllMessages();
+
+            assert.ok(taskManager1.queued(taskId), "Task manager 1 should be queued");
+            assert.ok(taskManager1.assignedTask(taskId), "Task manager 1 is not assigned task");
+            assert.ok(taskManager2.queued(taskId), "Task manager 2 should be queued");
+            assert.ok(!taskManager2.assignedTask(taskId), "Task manager 2 should not be assigned");
+
+            taskManager1.abandon(taskId);
+            containerRuntimeFactory.processAllMessages();
```

Comment:
I'm guessing after this processAllMessages, you want to verify that taskManager2 is assigned the task?